shell/key-bindings.fish

```diff
@@ -84,4 +86,26 @@ function fzf_key_bindings
     bind -M insert \cr fzf-history-widget
     bind -M insert \ec fzf-cd-widget
   end
+
+  function __fzf_get_starting_dir
+    set -l dir (commandline -t)
+
+    # Default to current dir if no argument input
+    if [ -z "$dir" ]
+        set dir '.'
+    end
+
+    # The commandline token might be escaped, we need to unescape it.
+    set dir (eval "printf '%s' $dir")
+
+    if [ ! -d "$dir" ]
+      return 1
```

Comment:
I like the idea. So we would be looking for the longest substring that is an existing path, the rest of the token is passed as query to `fzf`. These are some test cases to document my understanding:

Command line token | Test case | `find` + `fzf` behaviour
-- | -- | --
../exist/exist | whole path exists | `find ../exist/exist \| fzf`
../exist/not_exist/not_exist | first part of path exists  | `find ../exist \| fzf --query 'not_exist/not_exist'`
not_exist/non_exist | not a path, first part does not exist | `find . \| fzf --query not_exist/non_exist`
/not_exist/non_exist | not a path, starts with / | `find / \| fzf --query not_exist/non_exist`

The code logic would be:
1. Get current command line token
2. Check if it is an existing directory. If not, run `dirname` and check again. Repeat until you find a directory that exists. This will be `$dir`.
3. The remainder of token will be used as `--query` input.

Fish scripting has [commandline](https://fishshell.com/docs/current/commands.html#commandline) builtin which allows for reading and editing command line. From there, you have the usual tools to do string manipulation and file path checks. This is all doable, seems to be reasonably straightforward.

P.S. I appreciate your feedback and attention to detail regarding this. I like fish and fzf and hoped to contribute back with this PR. Due to the